### PR TITLE
Brand-consolidation - Integrate the platform brand-consolidation code from master

### PR DESCRIPTION
**Pull request by ncksllvn**
## Description
This PR continues on the work to merge `brand-consolidation` into `master`, with the most important step being a reliable feature flag and platform code based on that. This code has already been implemented into master at `src/platform/brand-consolidation`. However, it couldn't be used in the `brand-consolidation` branch until we reconciled its build script with `master`. That work is being merged here - department-of-veterans-affairs/vets-website/pull/8303 - and once it's merged, the code in this PR will be integrated throughout the `brand-consolidation` branch to enable a safe merge to `master`.

#### Unique to this PR
I ran `git checkout master src/platform/brand-consolidation` from the `brand-consolidation` branch to generate these changes.

### CODE REVIEW

File: src/platform/brand-consolidation/feature-flag.js [hunk 1]
```diff
@@ -0,0 +1,8 @@
+/**
+ * Returns whether the current Webpack build is executing as the brand-consolidation of Vets.gov/VA.gov.
+ * @returns {boolean}
+ * @module platform/brand-consolidation/feature-flag
+ */
+export default function isBrandConsolidationEnabled() {
+  return window.settings.brandConsolidationEnabled;
+}
```
Comment (ncksllvn) [inline src/platform/brand-consolidation/feature-flag.js]: @U-DON This is the module/feature flag for checking brand-consolidation builds. I'm hoping to have it integrated throughout in the near future.

File: src/platform/brand-consolidation/containers/BrandConsolidation.jsx [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+function BrandConsolidation({ brandConsolidationEnabled, children }) {
+  if (brandConsolidationEnabled) {
+    return <div data-brand-consolidation-enabled>{children}</div>;
+  }
+  return null;
+}
+
+export function mapStateToProps(state) {
+  return {
+    brandConsolidationEnabled: state.buildSettings.brandConsolidationEnabled
```
Comment (U-DON) [inline src/platform/brand-consolidation/containers/BrandConsolidation.jsx]: Where is the `buildSettings` prop coming from?

**Review by U-DON — COMMENTED** [review]
I think this looks good, but I'm unsure of how these components are intended to be used.

File: src/platform/brand-consolidation/containers/BrandConsolidation.jsx (hunk 2 above)
Comment (ncksllvn) [reply src/platform/brand-consolidation/containers/BrandConsolidation.jsx]: There's a directory, `platform/monitoring/BuildSettings` that contains a reducer that moves that data into the store. The data itself is created from the `window.settings` property, from the `js/settings.js` file. The reason we did it this way is that eventually the content will have different build cycle than the app code, so we need a way to pass data from the content to the code. This was the easiest way I could think to do that.

File: src/platform/brand-consolidation/containers/VetsDotGov.jsx [hunk 3]
```diff
@@ -0,0 +1,24 @@
+import React from 'react';
+import { connect } from 'react-redux';
+
+function VetsDotGov({ vetsDotGovEnabled, children }) {
+  if (vetsDotGovEnabled) {
+    return <div data-brand-consolidation-enabled>{children}</div>;
```
[the remark this answers is not in the document]
Comment (ncksllvn) [reply src/platform/brand-consolidation/containers/VetsDotGov.jsx]: Yeah, it was just there as a convenient selector for E2E/unit tests down the road.

**Comment by ncksllvn** [comment]
Thanks @U-DON ! Originally I wanted to have the brand-consolidation feature flag stored in state, rather than a raw lookup, so they aren't as useful as I was originally intending but I imagine they'll still have some usefulness.

**Review by U-DON — APPROVED** [review]

**Pull request merged**